pandas/core/internals/construction.py

```diff
@@ -349,14 +350,19 @@ def ndarray_to_mgr(
         if dtype is None and is_object_dtype(values.dtype):
             arrays = [
                 ensure_wrapped_if_datetimelike(
-                    maybe_infer_to_datetimelike(values[:, i])
+                    maybe_infer_to_datetimelike(values[:, i].copy())
                 )
                 for i in range(values.shape[1])
             ]
+            if copy:
```

Comment:
i would move the copy *after* the main if/else here for clarify

Reply:
Yes, good idea, updated